packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -36,6 +36,11 @@ class HotRunnerConfig {
   Future<bool> setupHotRestart() async {
     return true;
   }
+  /// A hook for implementations to perform any necessary operations right
+  /// before the runner is about to be shut down.
+  Future<void> runPreShutdownOperations() async {
```

Comment:
Please add a test case to `hot_test.dart` to ensure this gets called.

Reply:
Added a couple of tests